### DISCUSSION TITLE
Check if a user exists before updating through hooks

### DISCUSSION THREAD

**Discussion by magnusskare** [pocketbase/pocketbase#4617] [Q&A] [answered]
I'm working on a cron which should loop over some records and update them if required, based on some external conditions.

In a reduced example, this is what I'm doing:

```
// main.pb.js

data.forEach( item => {
    let userEmail = item.email;
    let newValue = item.value;

    let record = $app.dao().findFirstRecordByData("users", "email", userEmail);
    // If this fails, the loop is broken, but it should ideally continue

    if (record) {
        record.set("fieldToUpdate", newValue);
        $app.dao().saveRecord(record);
    }
});
```

This works well if the user actually exists, but as this is inside a ```forEach```, as soon as record returns zero users, it will exit the loop. What would the correct way to handle errors gracefully, so that the loop continues? In its current state, we never make it to the if-statement if ```record``` does not return a record. Am I going about this the wrong way?

**Comment by ganigeorgiev** [comment, marked as answer]
`findFirstRecordByData` method (and every other single record retrieval method) throws an error in case no item is found, so the check `if (record)` in this case is unnecessary. Or in other words, you could try:

```js
data.forEach( item => {
    try {
        let userEmail = item.email;
        let newValue = item.value;

        let record = $app.dao().findFirstRecordByData("users", "email", userEmail);

        record.set("fieldToUpdate", newValue);
        $app.dao().saveRecord(record);
    } catch (err) {
        // ignore or eventually log
        // https://pocketbase.io/docs/js-logging/
        $app.logger().debug("missing record or failed to save", "error", err)
    }
});
```

**Reply by magnusskare** [reply]
This works really well, thank you for the clarification.
